loki.source.doker: port latest Promtail changes (#5570)
grafana/agent: docs/sources/flow/reference/components/loki.source.docker.md

`forward_to`    | `list(LogsReceiver)` | List of receivers to send log entries to. | | yes
`labels`        | `map(string)`        | The default set of labels to apply on entries. | `"{}"` | no
`relabel_rules` | `RelabelRules`       | Relabeling rules to apply on log entries. | `"{}"` | no
`refresh_interval` | `duration`        | The refresh interval to use when connecting to the Docker daemon over HTTP(S). | `"60s"` | no

## Blocks

> The following blocks are supported inside the definition of `remote.http`:

Reviewer: ```suggestion
The following blocks are supported inside the definition of `loki.source.docker`:
```

--------- | ----- | ----------- | --------
client | [client][] | HTTP client settings when connecting to the endpoint. | no
client > basic_auth | [basic_auth][] | Configure basic_auth for authenticating to the endpoint. | no
client > authorization | [authorization][] | Configure generic authorization to the endpoint. | no
client > oauth2 | [oauth2][] | Configure OAuth2 for authenticating to the endpoint. | no
client > oauth2 > tls_config | [tls_config][] | Configure TLS settings for connecting to the endpoint. | no
client > tls_config | [tls_config][] | Configure TLS settings for connecting to the endpoint. | no

The `>` symbol indicates deeper levels of nesting. For example, `client >
basic_auth` refers to an `basic_auth` block defined inside a `client` block.

[client]: #client-block
[basic_auth]: #basic_auth-block
[authorization]: #authorization-block
[oauth2]: #oauth2-block
[tls_config]: #tls_config-block

### client block

The `client` block configures settings used to connect to the HTTP
server.

{{< docs/shared lookup="flow/reference/components/http-client-config-block.md" source="agent" version="<AGENT VERSION>" >}}

### basic_auth block

> The `basic_auth` block configures basic authentication to use when polling the
> configured URL.

Reviewer: ```suggestion
The `basic_auth` block configures basic authentication to use when talking to Docker.
```
This repeats below too.

--------- | ----- | ----------- | --------
client | [client][] | HTTP client settings when connecting to the endpoint. | no
client > basic_auth | [basic_auth][] | Configure basic_auth for authenticating to the endpoint. | no
client > authorization | [authorization][] | Configure generic authorization to the endpoint. | no
client > oauth2 | [oauth2][] | Configure OAuth2 for authenticating to the endpoint. | no
client > oauth2 > tls_config | [tls_config][] | Configure TLS settings for connecting to the endpoint. | no
client > tls_config | [tls_config][] | Configure TLS settings for connecting to the endpoint. | no

The `>` symbol indicates deeper levels of nesting. For example, `client >
basic_auth` refers to an `basic_auth` block defined inside a `client` block.

These blocks are only applicable when connecting to a Docker daemon over HTTP
or HTTPS and has no effect when connecting via a `unix:///` socket

[client]: #client-block
[basic_auth]: #basic_auth-block
[authorization]: #authorization-block
[oauth2]: #oauth2-block
[tls_config]: #tls_config-block

### client block

The `client` block configures settings used to connect to HTTP(S) Docker
daemons.

> {{< docs/shared lookup="flow/reference/components/http-client-config-block.md" source="agent" version="<AGENT VERSION>" >}}

Reviewer 2: ```suggestion
{{< docs/shared lookup="flow/reference/components/http-client-config-block.md" source="agent" version="<AGENT_VERSION>" >}}
```

### basic_auth block

The `basic_auth` block configures basic authentication for HTTP(S) Docker
daemons.

> {{< docs/shared lookup="flow/reference/components/basic-auth-block.md" source="agent" version="<AGENT VERSION>" >}}

Reviewer 2: ```suggestion
{{< docs/shared lookup="flow/reference/components/basic-auth-block.md" source="agent" version="<AGENT_VERSION>" >}}
```

### authorization block

The `authorization` block configures custom authorization to use for the Docker
daemon.

> {{< docs/shared lookup="flow/reference/components/authorization-block.md" source="agent" version="<AGENT VERSION>" >}}

Reviewer 2: ```suggestion
{{< docs/shared lookup="flow/reference/components/authorization-block.md" source="agent" version="<AGENT_VERSION>" >}}
```

### oauth2 block

The `oauth2` block configures OAuth2 authorization to use for the Docker
daemon.

> {{< docs/shared lookup="flow/reference/components/oauth2-block.md" source="agent" version="<AGENT VERSION>" >}}

Reviewer 2: ```suggestion
{{< docs/shared lookup="flow/reference/components/oauth2-block.md" source="agent" version="<AGENT_VERSION>" >}}
```

### tls_config block

The `tls_config` block configures TLS settings for connecting to HTTPS Docker
daemons.

> {{< docs/shared lookup="flow/reference/components/tls-config-block.md" source="agent" version="<AGENT VERSION>" >}}

Reviewer 2: ```suggestion
{{< docs/shared lookup="flow/reference/components/tls-config-block.md" source="agent" version="<AGENT_VERSION>" >}}
```
